packages/sync-actions/src/utils/copy-empty-array-props.js

```diff
@@ -1,30 +1,37 @@
+import isNil from 'lodash.isnil'
 /**
  * @function copyEmptyArrayProps
  * @description Takes two objects and if there are Array props in oldObj which doesn't exist in newObj, then copy it with an empty value.
  * @param {Object} oldObj
  * @param {Object} newObj
  * @returns {Array} Ordered Array [oldObj, newObj]
  */
-export default function copyEmptyArrayProps(oldObj, newObj) {
-  const newObjWithFixedEmptyArray = Object.entries(oldObj).reduce(
-    (acc, [key, value]) => {
-      const isArray = Array.isArray(value)
-      const newObjectValueIsUndefined = newObj[key] === undefined
+export default function copyEmptyArrayProps(oldObj = {}, newObj = {}) {
+  if (!isNil(oldObj) && !isNil(newObj)) {
+    const nextObjectWithEmptyArray = Object.entries(oldObj).reduce(
+      (nextObject, [key, value]) => {
+        const isArray = Array.isArray(value)
 
-      if (isArray && newObjectValueIsUndefined) {
-        return { ...acc, [key]: [] }
-      }
+        if (isArray && isNil(newObj[key])) {
+          return { ...nextObject, [key]: [] }
+        }
 
-      if (!isArray && typeof value === 'object' && newObj[key]) {
-        // recursion, so we could check for nested objects
-        const returnedNewObjCopy = copyEmptyArrayProps(value, newObj[key])[1]
-        return { ...acc, [key]: returnedNewObjCopy }
-      }
+        if (
+          !isNil(value) &&
+          !isArray &&
+          typeof value === 'object' &&
+          !isNil(newObj[key])
+        ) {
```

Comment:
☝🏽
this if statement doesn't make much sense and someone has to explain it...

- we check if `newObject[key]` is null or undefined.
- we check if `value` is null or undefined.

`value` derives from `newObject[key]`,because we use entries with the `newObj` as entry point

Reply:
hmm, I'm reading it now and see that the `value` check makes sense since we only want to go one level deeper if its an object in the `oldObj`.

The check for `newObject[key]` then makes sure that it's actually undefined. Since if it's already defined we shouldn't copy the values from `oldObj` and instead allow the new values to take priority.

Would above make sense for you too, @adnasa?

Then my follow-up question is, are we allowed to set it to `null` in the CTP? If so, we should keep the first check and not replace it with `isNil`

Reply:
I have no context here, so I'm not sure how to proceed on this topic. 
it's separate to me, and the scope of this PR is to have the null/undefined check

Reply:
it were introduced to category, fixing this issue [1439](https://github.com/commercetools/nodejs/issues/1439)